rewrite-java/src/main/java/org/openrewrite/java/UseGettersAndSetters.java

```diff
@@ -0,0 +1,115 @@
+package org.openrewrite.java;
+
+import org.openrewrite.Formatting;
+import org.openrewrite.Tree;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.List;
+
+import static org.openrewrite.internal.StringUtils.capitalize;
+
+public class UseGettersAndSetters extends JavaIsoRefactorVisitor {
+
+    private JavaType.Class classType;
+
+    public UseGettersAndSetters() {
+        setCursoringOn();
+    }
+
+    public void setClassType(JavaType.Class classType) {
+        this.classType = classType;
+    }
+
+    @Override
+    public J.VariableDecls.NamedVar visitVariable(J.VariableDecls.NamedVar variable) {
+        if (variable.isField(getCursor())) {
+            J.VariableDecls variableDecl = getCursor().firstEnclosing(J.VariableDecls.class);
+            if (variableDecl != null && J.Modifier.hasModifier(variableDecl.getModifiers(), "public")) {
+                String fieldName = variable.getSimpleName();
+                andThen(new ScopedPrivateField(variableDecl));
+                andThen(new GenerateGetter.Scoped(enclosingClass(), fieldName));
+                andThen(new GenerateSetter.Scoped(enclosingClass(), fieldName));
+                andThen(new ScopedUpdateReferences(fieldName, classType));
+            }
+        }
+        return super.visitVariable(variable);
+    }
+
+    public static class ScopedPrivateField extends JavaIsoRefactorVisitor {
```

Comment:
I'd probably leave this and UpdateReferences and ScopedAssign private since I consider them to be implementation details of this class rather than a public API it's providing. Also there's no particular need to prefix their names with "Scoped".